### PR TITLE
Fix duplicate check in hb_ot_rotate chars

### LINKED ISSUE #2099: Duplicate c->font->has_glyph (codepoint) check in hb_rotate_chars
Looks like there's a duplicate line after the condition was rewritten.

### PR DESCRIPTION
**Pull request by drott**
Fixes #2099. Fix indentation in mirroring section as well.

### CODE REVIEW

File: src/hb-ot-shape.cc [hunk 1]
```diff
@@ -642,19 +642,18 @@ hb_ot_rotate_chars (const hb_ot_shape_context_t *c)
     for (unsigned int i = 0; i < count; i++) {
       hb_codepoint_t codepoint = unicode->mirroring (info[i].codepoint);
       if (unlikely (codepoint != info[i].codepoint && c->font->has_glyph (codepoint)))
-	info[i].codepoint = codepoint;
+        info[i].codepoint = codepoint;
       else
-	info[i].mask |= rtlm_mask;
+        info[i].mask |= rtlm_mask;
```
Comment (ebraminio) [inline src/hb-ot-shape.cc]: Please revert turning tab to spaces, thanks :)

Comment (drott) [reply src/hb-ot-shape.cc]: It seems to me that the rest of the file is space indented. Or was this meant to be ironic?

Comment (ebraminio) [reply src/hb-ot-shape.cc]: Rest of the file is mostly consistent on HarfBuzz's current default style that is using two spaces for indention but turning 8 of them to tab, not that fan of the style personally but let preserve it for now :)

Comment (drott) [reply src/hb-ot-shape.cc]: 🤯Wow, I've never seen such a convention, hadn't even noticed that in HarfBuzz. What's the origin of that convention or why would it ever make sense? :) - but yes, updated the CL.

Comment (ebraminio) [reply src/hb-ot-shape.cc]: @behdad should know :)

**Pull request merged**